project/deps.sc

```diff
@@ -169,8 +169,13 @@ object Versions {
   val `zio-schema`    = "0.4.12"
 }
 
+object DevMode {
+  val devMode: Boolean = System.getenv("MORPHIR_SCALA_DEV_MODE") == "true"
+}
+
 object ScalaVersions {
-  val all      = Seq(scala213, scala3x)
+  import DevMode._
+  val all      = if (devMode) Seq(scala3x) else Seq(scala213, scala3x)
```

Comment:
Take a look here: http://mill-build.com/mill/example/cross/9-dynamic-cross-modules.html

For something like this we should be making use of `interp.watchValue`